pkg/bisect/bisect.go

```diff
@@ -460,25 +464,51 @@ func (env *env) test() (*testResult, error) {
 		}
 		return res, nil
 	}
-	testStart := time.Now()
-	results, err := env.inst.Test(NumTests, cfg.Repro.Syz, cfg.Repro.Opts, cfg.Repro.C)
-	env.testTime += time.Since(testStart)
-	if err != nil {
-		env.log("failed: %v", err)
-		return res, nil
-	}
-	bad, good, rep := env.processResults(current, results)
-	res.rep = rep
-	res.verdict = vcs.BisectSkip
-	if bad != 0 {
-		res.verdict = vcs.BisectBad
-	} else if NumTests-good-bad > NumTests/3*2 {
-		// More than 2/3 of instances failed with infrastructure error,
-		// can't reliably tell that the commit is good.
-		res.verdict = vcs.BisectSkip
-	} else if good != 0 {
-		res.verdict = vcs.BisectGood
+
+	// Increase number of tests in case reproducer is flaky.
```

Comment:
I think the logic here is correct, but I found it somewhat hard to understand down to all details related to the second iteration (when we do second iteration and when we don't, etc). It's also better to keep numTests increment and check together to avoid unsynchronization in future (say, if increment moved after the check).
I think it can be simplified if we always do 2x tests for the initial iteration as:
```
	numTests := NumTests
	if env.flaky || env.numTests == 0 {
		// Use twice as many instances if the bug is flaky and during initial testing
		// (as we don't know yet if it's flaky or not).
		numTests *= 2
	}
	env.numTests++
```
Then we keep the rest of the logic as is (except for setting flaky flag).